src/comestible_inv_area.cpp

```diff
@@ -0,0 +1,275 @@
+}
+
+inline std::string comestible_inv_area::get_location_key()
+{
+    return comestible_inv_area::get_info( get_relative_location() ).minimapname;
+}
+
+template <typename T>
+static comestible_inv_area::area_items mi_stacked( T items )
+{
+    //create a new container for our stacked items
+    comestible_inv_area::area_items stacks;
+    //    // make a list of the items first, so we can add non stacked items back on
+    //    std::list<item> items(things.begin(), things.end());
+    // used to recall indices we stored `itype_id' item at in itemstack
+    std::unordered_map<itype_id, std::set<int>> cache;
+    // iterate through and create stacks
+    for( auto &elem : items ) {
+        const auto id = elem.typeId();
+        auto iter = cache.find( id );
+        bool got_stacked = false;
+        // cache entry exists
+        if( iter != cache.end() ) {
+            // check to see if it stacks with each item in a stack, not just front()
+            for( auto &idx : iter->second ) {
+                for( auto &it : stacks[idx] ) {
+                    if( ( got_stacked = it->display_stacked_with( elem ) ) ) {
+                        stacks[idx].push_back( &elem );
+                        break;
+                    }
+                }
+                if( got_stacked ) {
+                    break;
+                }
+            }
+        }
+        if( !got_stacked ) {
+            cache[id].insert( stacks.size() );
+            stacks.push_back( { &elem } );
+        }
+    }
+    return stacks;
+}
+
+comestible_inv_area::area_items comestible_inv_area::get_items( bool from_cargo )
+{
+    player &u = g->u;
+    comestible_inv_area::area_items retval;
+
+    if( info.id == comestible_inv_area_info::AIM_INVENTORY ) {
+        const invslice &stacks = u.inv.slice();
+
+        retval.reserve( stacks.size() );
+        for( auto stack : stacks ) {
+            std::list<item *> item_pointers;
+            for( item &i : *stack ) {
+                item_pointers.push_back( &i );
+            }
+
+            retval.push_back( item_pointers );
+        }
+    } else if( info.id == comestible_inv_area_info::AIM_WORN ) {
+        auto iter = u.worn.begin();
+
+        retval.reserve( u.worn.size() );
+        for( size_t i = 0; i < u.worn.size(); ++i, ++iter ) {
+            std::list<item *> item_pointers = { & *iter };
+            retval.push_back( item_pointers );
+        }
+    } else {
+        if( from_cargo ) {
+            assert( has_vehicle() );
+            retval = mi_stacked( veh->get_items( veh_part ) );
+        } else {
+            retval = mi_stacked( g->m.i_at( g->u.pos() + offset ) );
+        }
+    }
+
+    return retval;
+}
+
+units::volume comestible_inv_area::get_max_volume( bool use_vehicle )
+{
+    if( use_vehicle ) {
+        assert( has_vehicle() );
+        return veh->max_volume( veh_part );
+    } else {
+        map &m = g->m;
+        return m.max_volume( g->u.pos() + offset );
```

Comment:
```suggestion
        return g->m.max_volume( g->u.pos() + offset );
```
And remove the definition of `m` above.